src/Integration.UnitTests/Binding/RoslynProjectBinderTests.cs

```diff
@@ -0,0 +1,213 @@
+            bindingOperationMock.Verify(x=> x.Commit(), Times.Never);
+
+            bindAction();
+
+            bindingOperationMock.Verify(x => x.Commit(), Times.Once);
+        }
+
+        [TestMethod]
+        public void IsBound_RulesetFileDoesNotExist_False()
+        {
+            var bindingConfiguration = GetBindingConfiguration();
+            var projectMock = GetCSharpProject();
+            var mockRulesetPath = SetupRulesetPath();
+
+            fileSystemMock.Setup(x => x.File.Exists(mockRulesetPath)).Returns(false);
+
+            var result = testSubject.IsBound(bindingConfiguration, projectMock);
+            result.Should().BeFalse();
+        }
+
+        [TestMethod]
+        public void IsBound_RulesetFileExistsButFailsToLoad_False()
+        {
+            var bindingConfiguration = GetBindingConfiguration();
+            var projectMock = GetCSharpProject();
+            var mockRulesetPath = SetupRulesetPath();
+
+            fileSystemMock.Setup(x => x.File.Exists(mockRulesetPath)).Returns(true);
+            ruleSetSerializerMock.Setup(x => x.LoadRuleSet(mockRulesetPath)).Returns(null as RuleSet);
+
+            var result = testSubject.IsBound(bindingConfiguration, projectMock);
+            result.Should().BeFalse();
+        }
+
+        [TestMethod]
+        public void IsBound_RulesetFileHasNoDeclarations_False()
+        {
+            var bindingConfiguration = GetBindingConfiguration();
+            var projectMock = GetCSharpProject();
+            var mockRulesetPath = SetupRulesetPath();
+
+            fileSystemMock.Setup(x => x.File.Exists(mockRulesetPath)).Returns(true);
+            
+            ruleSetSerializerMock.Setup(x => x.LoadRuleSet(mockRulesetPath)).Returns(new RuleSet("test"));
+
+            solutionRuleSetsInformationProviderMock
+                .Setup(x => x.GetProjectRuleSetsDeclarations(projectMock))
+                .Returns(Array.Empty<RuleSetDeclaration>());
+
+            var result = testSubject.IsBound(bindingConfiguration, projectMock);
+            result.Should().BeFalse();
+        }
+
+        [TestMethod]
+        public void IsBound_RulesetFileHasOneDeclaration_CantLoadProjectRulesetFile_False()
+        {
+            var bindingConfiguration = GetBindingConfiguration();
+            var projectMock = GetCSharpProject();
+            var mockRulesetPath = SetupRulesetPath();
+
+            fileSystemMock.Setup(x => x.File.Exists(mockRulesetPath)).Returns(true);
+
+            ruleSetSerializerMock.Setup(x => x.LoadRuleSet(mockRulesetPath)).Returns(new RuleSet("test"));
+
+            var ruleSetDeclaration = GetRuleSetDeclaration(projectMock);
+
+            solutionRuleSetsInformationProviderMock
+                .Setup(x => x.GetProjectRuleSetsDeclarations(projectMock))
+                .Returns(new List<RuleSetDeclaration> {ruleSetDeclaration});
+
+            var filePath = "";
+            solutionRuleSetsInformationProviderMock
```

Comment:
The `IsBoundXXX_False` tests are a linked set of tests that check a series of conditions in the order that they are called in the product code, which makes sense.
However, this means that there's a lot of duplication in the mock setup code, and also the mock setup is pretty ugly, which makes it hard to see the commonality and the differences. It would be easy to miss a copy-paste error.

The tests also mock a lot of things, then only check final return value. My immediate thought on reading such tests is "is the test returning `false` for the reason I think it is? Or is there a bug in the test setup, so I'm getting the right answer for the wrong reason?". I normally take one of two approaches. The first is to debug through each test when writing it to make sure it fails for the right reason. That tells me the tests are correct at the of writing, but doesn't help when the code changes in the future. The second is to make the test self-checking e.g. if the test expects to the first condition to fail, then assert that the second condition wasn't checked. It's more effort, but it gives more confidence that a test is passing for the right reason. The tests already implicitly depend on the order of operations in the product code; adding assertions just makes the dependency explicit.